sdk/data_kusto/src/operations/query.rs

```diff
@@ -0,0 +1,195 @@
+use crate::client::KustoClient;
+use async_convert::{async_trait, TryFrom};
+use azure_core::prelude::*;
+use azure_core::setters;
+use azure_core::{collect_pinned_stream, Response as HttpResponse};
+use futures::future::BoxFuture;
+use serde::{Deserialize, Serialize};
+
+type ExecuteQuery = BoxFuture<'static, crate::error::Result<KustoResponseDataSetV2>>;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct QueryBody {
+    /// Name of the database in scope that is the target of the query or control command
+    db: String,
+    /// Text of the query or control command to execute
+    csl: String,
+}
+
+#[derive(Debug, Clone)]
+pub struct ExecuteQueryBuilder {
+    client: KustoClient,
+    database: String,
+    query: String,
+    client_request_id: Option<ClientRequestId>,
+    app: Option<App>,
+    user: Option<User>,
+    context: Context,
+}
+
+impl ExecuteQueryBuilder {
+    pub(crate) fn new(
+        client: KustoClient,
+        database: String,
+        query: String,
+        context: Context,
+    ) -> Self {
+        Self {
+            client,
+            database,
+            query: query.trim().into(),
+            client_request_id: None,
+            app: None,
+            user: None,
+            context,
+        }
+    }
+
+    setters! {
+        client_request_id: ClientRequestId => Some(client_request_id),
+        app: App => Some(app),
+        user: User => Some(user),
+        query: String => query,
+        database: String => database,
+        context: Context => context,
+    }
+
+    pub fn into_future(self) -> ExecuteQuery {
+        let this = self.clone();
+        let ctx = self.context.clone();
+
+        Box::pin(async move {
+            let url = this.client.query_url();
+            let mut request = this.client.prepare_request(url, http::Method::POST);
+
+            if let Some(request_id) = &this.client_request_id {
+                request.insert_headers(request_id);
+            };
+            if let Some(app) = &this.app {
+                request.insert_headers(app);
+            };
+            if let Some(user) = &this.user {
+                request.insert_headers(user);
+            };
+
+            let body = QueryBody {
+                db: this.database,
+                csl: this.query,
+            };
+            let bytes = bytes::Bytes::from(serde_json::to_string(&body)?);
+            request.insert_headers(&ContentLength::new(bytes.len() as i32));
+            request.set_body(bytes.into());
+
+            let response = self
+                .client
+                .pipeline()
+                .send(&mut ctx.clone(), &mut request)
+                .await?;
+
+            <KustoResponseDataSetV2 as TryFrom<HttpResponse>>::try_from(response).await
```

Comment:
@yoshuawuyts - I tried using your `async-convert` crate here to eventually make this builder generic over the response type. The management and query request builders would likely look more or less the same, and differ only in the response type. I wanted to end up up with something like

```rust
response.try_into().await
```

but I am doing something wrong. Or am completely off track :). Could you give us a hint if this is the right place to use it, and if so, how?